tests/unit/test_operator.py

```diff
@@ -1,81 +1,104 @@
+    rel_data = harness.model.get_relation(RELATION_NAME, rel_id).data[harness.model.app]
 
+    # The GRPC port from the default config and the GRPC_SVC_NAME set in the charm code
+    # are the expected values of this test case
+    expected = {"port": harness.model.config["grpc-port"], "name": GRPC_SVC_NAME}
+    assert rel_data["port"] == expected["port"]
+    assert rel_data["name"] == expected["name"]
 
-def test_no_relation(harness):
+
+def test_kubernetes_component_created(harness, mocked_lightkube_client):
+    """Test that Kubernetes component is created when we have leadership."""
+    # Needed because the kubernetes component will only apply to k8s if we are the leader
     harness.set_leader(True)
-    harness.add_oci_resource(
-        "oci-image",
-        {
-            "registrypath": "ci-test",
-            "username": "",
-            "password": "",
-        },
-    )
-    harness.begin_with_initial_hooks()
+    harness.begin()
+
+    # Need to mock the leadership-gate to be active, and the kubernetes auth component so that it
+    # sees the expected resources when calling _get_missing_kubernetes_resources
+    kubernetes_resources = harness.charm.kubernetes_resources
+    kubernetes_resources.component._get_missing_kubernetes_resources = MagicMock(return_value=[])
 
-    assert harness.charm.model.unit.status == ActiveStatus("")
+    harness.charm.on.install.emit()
 
+    assert isinstance(harness.charm.kubernetes_resources.status, ActiveStatus)
 
-def test_one_relation(harness):
+    # Assert that expected amount of apply calls were made
+    # This simulates the Kubernetes resources being created
+    assert mocked_lightkube_client.apply.call_count == 1
+
+
+def test_pebble_service_container_running(harness, mocked_lightkube_client):
+    """Test that the pebble service of the charm's mlmd-grpc-server container is running."""
     harness.set_leader(True)
-    harness.add_oci_resource(
-        "oci-image",
-        {
-            "registrypath": "ci-test",
-            "username": "",
-            "password": "",
-        },
-    )
-    rel_id = harness.add_relation("mysql", "mysql")
-    harness.add_relation_unit(rel_id, "mysql/0")
-    harness.update_relation_data(
-        rel_id,
-        "mysql/0",
-        {
-            "database": "unit-db",
-            "host": "unit-host",
-            "port": "unit-port",
-            "user": "unit-user",
-            "password": "unit-password",
-            "root_password": "unit-root",
-        },
+    harness.begin()
+    harness.set_can_connect(CONTAINER_NAME, True)
+
+    harness.charm.kubernetes_resources.get_status = MagicMock(return_value=ActiveStatus())
+
+    harness.charm.on.install.emit()
+
+    assert isinstance(harness.charm.unit.status, ActiveStatus)
+
+    container = harness.charm.unit.get_container(CONTAINER_NAME)
+    # Assert that sidecar container is up and its service is running
+    assert container.get_service(SERVICE_NAME).is_running()
```

Comment:
perhaps you can add a check on the pebble layer's contents as well

Reply:
Since the pebble layer is very simple and only the command could change, do you think it would add value to the tests? I could check for the command, but I'm not sure if we'd want to do it. Let me know what you think (: